src/components/CodeSnippet/CodeBlock.tsx

```diff
@@ -0,0 +1,56 @@
+import { useState } from "react";
+import copyCode from "./CopyCode";
+import customTheme from "./CustomTheme";
+import { CodeBlock} from 'react-code-blocks';
+import styles from './codeBlock.module.scss';
+import { copyIcon } from "./CopyIcon";
+
+//Documentation for react-code-blocks: https://www.npmjs.com/package/react-code-blocks
+
+interface CodeBlockProps {
+    language: string;
+    code: string;
+
+  }
+
+export function CodeSample(props: CodeBlockProps): JSX.Element{
+
+    const [buttonText, setButtonText] = useState<string>('Copy');
+    const [showIcon, setShowIcon] = useState<boolean>(true);
+
+    const handleCopyClick = () => {
+      copyCode(props.code);
+      setButtonText('Copied!');
+      setShowIcon(false);
+      setTimeout(() => {
```

Comment:
Don't use a timeout or change the text of the button, use a toast instead (from the react component library)